bundles/org.openhab.core/src/main/java/org/openhab/core/internal/service/WatchServiceImpl.java

```diff
@@ -230,7 +230,9 @@ public void onEvent(@Nullable DirectoryChangeEvent directoryChangeEvent) throws
                 future.cancel(true);
             }
             future = scheduler.schedule(() -> notifyListeners(path), PROCESSING_TIME, TimeUnit.MILLISECONDS);
-            scheduledEventKinds.computeIfAbsent(path, k -> new CopyOnWriteArrayList<>()).add(directoryChangeEvent);
+            List<DirectoryChangeEvent> list = Objects
+                    .requireNonNull(scheduledEventKinds.computeIfAbsent(path, k -> new CopyOnWriteArrayList<>()));
+            list.add(directoryChangeEvent);
```

Comment:
This is less readable and I would prefer to roll it back for now. I am fine will all other changes and would prefer to merge first.

Reply:
@holgerfriedrich : done